src/item_factory.h

```diff
@@ -285,6 +292,9 @@ class Item_factory
         std::unordered_map<itype_id, ammotype> migrated_ammo;
         std::unordered_map<itype_id, itype_id> migrated_magazines;
 
+        // cache for holsters
+        std::unordered_map<itype_id, std::vector<const itype *>> type_contained_in_holsters;
```

Comment:
`std::vector<std::reference_wrapper<const itype>>` would probably be a better fit than a `std::vector<const itype *>>`.

Reply:
Actually, `Item_factory::find` returns `std::vector<const itype *>`. So I would reinterpret it there as a reference.

It seems that whenever working with vector of itype, we use pointers. It might be that the code is old.

Do you still thing I should change it to refs?

Reply:
> It seems that whenever working with vector of itype, we use pointers. It might be that the code is old.

I think it is just that the code is old, and everyone who has used the code since then has just kept it as is. There isn't any performance cost to converting a pointer to a reference, so I think they should be stored as references.